draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2797,11 +2797,14 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
 
 <section title="Identifying HTTPS Servers" anchor="https.identify.server">
 <t>
-   In general, HTTP/TLS requests are generated by dereferencing a URI.
-   As a consequence, the hostname for the server is known to the client.
-   If the hostname is available, the client &MUST; check it against the
-   server's identity, as presented in the server's Certificate message,
-   to prevent an on-path attacker from substituting the key.
+   In general, HTTP/TLS requests are generated by dereferencing a URI. As a
+   consequence, the hostname for the server is known to the client. If the
+   hostname is available, the client MUST use the hostname as a reference
+   identity as input to the verification processed defined in Section 6 of
+   <xref target="RFC6125"/>. This process requires that a client be configured with a set of
+   trust anchors. Certificate verification is used to prevent server
+   impersonation by an on-path attacker or by an attacker that controls name
+   resolution.
```

Comment:
I feel like REQUIRES is going to be confused with the rfc2119 keyword REQUIRED. If that's intentional, we shouldn't be re-stating another spec's requirement.

Reply:
Leaving aside "REQUIRES" (on which I agree with Mark)...

Roy's suggestion here says a couple of things that need to be carefully reconciled.

* a URI which identifies an origin server (within the authority component)
* use that origin's host (as the reference identity)

I think that the fact that an origin server is identified using *something* from the authority component is absolutely perfect.  However, the jump to "host" is potentially risky.  I would stick to "The client MUST use the origin server identity from the URI as the reference identity for the verification process [...]".

I prefer the split of the last piece into two sentences (Mark's version) rather than having them joined, but that isn't substantive.

Reply:
yep, that's fine ... for some reason I read it as a requirement. +1 on Martin's suggestion.